api/v1/handlers.go

```diff
@@ -609,23 +602,22 @@ func (s *apiV1) handleAdd(c echo.Context, u *util.User) error {
 		}
 	}
 
-	content, err := s.CM.AddDatabaseTracking(ctx, u, dserv, nd.Cid(), filename, replication)
-	if err != nil {
-		return xerrors.Errorf("encountered problem computing object references: %w", err)
+	if err := util.DumpBlockstoreTo(ctx, s.tracer, bs, s.Node.Blockstore); err != nil {
+		s.log.Errorf("failed to move data from staging to main blockstore: %w", err)
```

Comment:
no need to log here since the return err is still logged downstream. maybe wrap the error with your context message

Reply:
Also, nice fix. We should really have the blocks before creating a db record. I was thinking about doing this later. I have a much better approach, but we can discuss it later after this PR

Reply:
can you add a comment that communicates that file uploads block objects will not be created by the pinner, since  `AddDatabaseTracking` is no longer used here